src/azul/deployment.py

```diff
@@ -434,8 +438,47 @@ def client(self, *args, **kwargs):
         Caching the result of this function is not necessary and will be harmful
         if the cached value is used by a thread other than the one that called
         this function.
+
+        :param azul_logging: Whether to log the client's requests and
+                             responses. Note that using DEBUG level will
+                             enable logging of request bodies, which could
+                             contain sensetive or secret informantion.
         """
-        return self.boto3_session.client(*args, **kwargs)
+        client = self.boto3_session.client(*args, **kwargs)
+        if azul_logging:
+            client.meta.events.register('before-send', self._log_client_request)
+            client.meta.events.register('response-received', self._log_client_response)
+        return client
+
+    def _log_client_request(self,
+                            event_name: str,
+                            request: AWSPreparedRequest,
+                            **kwargs: Any
+                            ) -> Optional[AWSResponse]:
+        event_name = self.__format_boto3_event_name(event_name, rm_prefix='before-send')
+        boto3_log.info('%s    Making %s request to %s',
+                       event_name,
+                       request.method,
+                       request.url)
+        boto3_log.info(http_body_log_message(boto3_log, 'request', request.body))
+        return None
+
+    def __format_boto3_event_name(self, event_name: str, rm_prefix: str) -> str:
+        rm_prefix += '.'
```

Comment:
Whenever you have a separator character, `partition` is usually more straight forward.